src/Microsoft.DocAsCode.MarkdownLite/SourceInfo.cs

```diff
@@ -18,14 +19,33 @@ private SourceInfo(string markdown, string file, int lineNumber)
 
         public int LineNumber { get; }
 
+        public int ValidLineCount { get; }
+
         public static SourceInfo Create(string markdown, string file, int lineNumber = 1)
         {
-            return new SourceInfo(markdown, file, lineNumber);
+            return new SourceInfo(markdown, file, lineNumber, GetValidLineCount(markdown));
         }
 
         public SourceInfo Copy(string markdown, int lineOffset = 0)
         {
-            return new SourceInfo(markdown, File, LineNumber + lineOffset);
+            return new SourceInfo(markdown, File, LineNumber + lineOffset, GetValidLineCount(markdown));
+        }
+
+        private static int GetValidLineCount(string markdown)
+        {
+            var indexOfLastChar = markdown.Length - 1;
+            var validLineCount = 0;
+
+            while (indexOfLastChar >= 0 && markdown[indexOfLastChar] == '\n')
```

Comment:
bad performance, it will create new string.

---
In reply to: [76751491](https://github.com/dotnet/docfx/pull/646#discussion_r76751491) [](ancestors = 76751491,76751147)